python/pyspark/sql/tests/connect/test_connect_basic.py

```diff
@@ -465,6 +466,29 @@ def test_fill_na(self):
             self.spark.sql(query).na.fill({"a": True, "b": 2}).toPandas(),
         )
 
+    def test_with_columns(self):
+        # SPARK-41256: test withColumn(s).
+        self.assert_eq(
```

Comment:
Does equality work like this for the pandas DFs? I'm wondering if this should be `pdf.equals(other)`?

Reply:
I believe test assert is more powerful. If you encounter a false-equal case this assert prints much more information to help understand where is the in-quality.  

`assert pdf.equals(other)` will just see these two are not equal, that's all. see https://github.com/apache/spark/pull/38670

Reply:
SG